Dart/src/com/jetbrains/lang/dart/ide/runner/server/webdev/DartWebdevConfigurationProducer.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.jetbrains.lang.dart.ide.runner.server.webdev;
+
+import com.intellij.execution.actions.ConfigurationContext;
+import com.intellij.execution.actions.LazyRunConfigurationProducer;
+import com.intellij.execution.configurations.ConfigurationFactory;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ProjectFileIndex;
+import com.intellij.openapi.roots.ProjectRootManager;
+import com.intellij.openapi.util.Ref;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.impl.source.html.HtmlLikeFile;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import com.jetbrains.lang.dart.sdk.DartSdk;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class DartWebdevConfigurationProducer extends LazyRunConfigurationProducer<DartWebdevConfiguration> {
+  @NotNull
+  @Override
+  public ConfigurationFactory getConfigurationFactory() {
+    return DartWebdevConfigurationType.getInstance().getConfigurationFactories()[0];
+  }
+
+  @Override
+  protected boolean setupConfigurationFromContext(final @NotNull DartWebdevConfiguration configuration,
+                                                  final @NotNull ConfigurationContext context,
+                                                  final @NotNull Ref<PsiElement> sourceElement) {
+    final VirtualFile htmlFile = getRunnableHtmlFileFromContext(context);
+    if (htmlFile == null) {
+      return false;
+    }
+
+    configuration.getParameters().setHtmlFilePath(htmlFile.getPath());
+    configuration.setGeneratedName();
+    sourceElement.set(sourceElement.isNull() ? null : sourceElement.get().getContainingFile());
+    return true;
+  }
+
+  @Override
+  public boolean isConfigurationFromContext(final @NotNull DartWebdevConfiguration configuration,
+                                            final @NotNull ConfigurationContext context) {
+    final VirtualFile htmlFile = getHtmlFileFromContext(context);
+    return htmlFile != null && htmlFile.getPath().equals(configuration.getParameters().getHtmlFilePath());
+  }
+
+  @Nullable
+  private static VirtualFile getRunnableHtmlFileFromContext(@NotNull final ConfigurationContext context) {
+    // return null if this is not a Dart project
+    final Project project = context.getProject();
+    if (project == null || !DartAnalysisServerService.getInstance(project).serverReadyForRequest()) {
```

Comment:
Just realized that `serverReadyForRequest()` tries to start the analysis server if it's not already running. And this line of code runs on any right click in any file in any project =).

I don't think we care about analysis server in RunConfigurationProducer. We care first of all about being able to start the produced run configuration right away.